File: 🟡 Medium/1605 - Find Valid Matrix Given Row and Column Sums/_1605FindValidMatrixGivenRowAndColumnSums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum,
                                      vector<int>& colSum) {
        int rowLen = rowSum.size();
        int colLen = colSum.size();

        vector<int> currRowSum(rowLen);
        vector<int> currColSum(colLen);

        vector<vector<int>> origMatrix(rowLen, vector<int>(colLen));
        for (int i = 0; i < rowLen; i++) {
            for (int j = 0; j < colLen; j++) {
                origMatrix[i][j] =
                    min(rowSum[i] - currRowSum[i], colSum[j] - currColSum[j]);

                currRowSum[i] += origMatrix[i][j];
                currColSum[j] += origMatrix[i][j];
            }
        }
        return origMatrix;
    }
};

File: 🟡 Medium/1605 - Find Valid Matrix Given Row and Column Sums/_1605FindValidMatrixGivenRowAndColumnSums.java
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int rowLen = rowSum.length;
        int colLen = colSum.length;

        int[] currRowSum = new int[rowLen];
        int[] currColSum = new int[colLen];

        int[][] origMatrix = new int[rowLen][colLen];
        for (int i = 0; i < rowLen; i++) {
            for (int j = 0; j < colLen; j++) {
                origMatrix[i][j] = Math.min(rowSum[i] - currRowSum[i], colSum[j] - currColSum[j]);

                currRowSum[i] += origMatrix[i][j];
                currColSum[j] += origMatrix[i][j];
            }
        }
        return origMatrix;
    }
}